[Bug]: Compodoc Fails To Start When Using PNPM (7.0.0-beta.44)

### Describe the bug

Related to Angular setup with compodoc when using pnpm package manager. The storybook will fail to start (no errors)

Debug logs show error:
> ERR_PNPM_NO_SCRIPT  Missing script: compodoc

The installer should either add this to package.json or allow for documentation to be imported in the code (manually).

### To Reproduce

```shell
*angular v.14
1. ng new angular-book
2. pnpx storybook@7.0.0-beta.44 init
3. Do you want to use Compodoc for documentation? Yes
4. pnpx run storybook
```


### System

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (16) x64 AMD Ryzen 7 5800X 8-Core Processor
  Binaries:
    Node: 16.18.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.19.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (96.0.1054.62)
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-beta.44 => 7.0.0-beta.44
    @storybook/addon-interactions: ^7.0.0-beta.44 => 7.0.0-beta.44
    @storybook/addon-links: ^7.0.0-beta.44 => 7.0.0-beta.44
    @storybook/angular: ^7.0.0-beta.44 => 7.0.0-beta.44
    @storybook/blocks: ^7.0.0-beta.44 => 7.0.0-beta.44
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1
```


### Additional context

works ok with npm

--- a/code/lib/cli/src/js-package-manager/PNPMProxy.test.ts
+++ b/code/lib/cli/src/js-package-manager/PNPMProxy.test.ts
@@ -54,7 +54,7 @@ describe('NPM Proxy', () => {
 
       expect(executeCommandSpy).toHaveBeenLastCalledWith(
         'pnpm',
-        ['run', 'compodoc', '-e', 'json', '-d', '.'],
+        ['exec', 'compodoc', '-e', 'json', '-d', '.'],
         undefined,
         undefined
       );

--- a/code/lib/cli/src/js-package-manager/PNPMProxy.ts
+++ b/code/lib/cli/src/js-package-manager/PNPMProxy.ts
@@ -25,7 +25,7 @@ export class PNPMProxy extends JsPackageManager {
   }
 
   runPackageCommand(command: string, args: string[], cwd?: string): string {
-    return this.executeCommand(`pnpm`, ['run', command, ...args], undefined, cwd);
+    return this.executeCommand(`pnpm`, ['exec', command, ...args], undefined, cwd);
   }
 
   protected getResolutions(packageJson: PackageJson, versions: Record<string, string>) {

